analyzers/src/SonarAnalyzer.Common/Metrics/MetricsBase.cs

```diff
@@ -119,6 +118,14 @@ public FileComments GetComments(bool ignoreHeaderComments)
         private IEnumerable<SyntaxNode> FunctionNodes =>
             tree.GetRoot().DescendantNodes().Where(IsFunction);
 
+        private FileLinePositionSpan? GetMappedLineSpan(SyntaxToken token) =>
+            !IsEndOfFile(token)
+            && token.GetLocation().GetMappedLineSpan() is { IsValid: true } mappedLineSpan
+            && (!GeneratedCodeRecognizer.IsRazor(token.SyntaxTree) || mappedLineSpan.HasMappedPath)
+            && IsInSameFile(mappedLineSpan)
```

Comment:
I assume that in razor files, if `mappedLineSpan.HasMappedPath` is `false`, `IsInSameFile(mappedLineSpan)` is `false` as well. Therefore the `mappedLineSpan.HasMappedPath` check might be redundant. Can you confirm this? If so, I would remove 
```suggestion
            && IsInSameFile(mappedLineSpan)
```
and add a comment about the `mappedLineSpan.HasMappedPath` being redundant.

Reply:
This is not redundant. For non-razor code, all the calls are returning false.

Reply:
I'm not sure I understand the comment.

I think the `HasMappedPath` is required only for Razor-generated code since it is mapped to the original, user-written one.

For non-Razor code, `HasMappedPath` returns `false`, which will result in skipped metrics for normal `.cs` files.

Reply:
You are right, but I would keep it since it is way faster to check this than to do string comparisons (although we cannot completely remove those).